qcodes/instrument_drivers/Andor/DU401.py

```diff
@@ -0,0 +1,424 @@
+        self.error_check(code, 'GetTemperatureRange')
+        return c_min_temp.value, c_max_temp.value
+
+    def initialize(self, directory):
+        code = self.dll.Initialize(directory)
+        self.error_check(code, 'Initialize')
+
+    def is_cooler_on(self):
+        c_cooler_status = ctypes.c_int()
+        code = self.dll.IsCoolerOn(ctypes.byref(c_cooler_status))
+        self.error_check(code, 'IsCoolerOn')
+        return c_cooler_status.value
+
+    def set_accumulation_cycle_time(self, cycle_time):
+        c_cycle_time = ctypes.c_float(cycle_time)
+        code = self.dll.SetAccumulationCycleTime(c_cycle_time)
+        self.error_check(code, 'SetAccumulationCycleTime')
+
+    def set_acqiusition_mode(self, mode):
+        c_mode = ctypes.c_int(mode)
+        code = self.dll.SetAcquisitionMode(c_mode)
+        self.error_check(code, 'SetAcquisitionMode')
+
+    def set_current_camera(self, camera_handle):
+        c_camera_handle = ctypes.c_long(camera_handle)
+        code = self.dll.SetCurrentCamera(c_camera_handle)
+        self.error_check(code, 'SetCurrentCamera')
+
+    def set_exposure_time(self, exposure_time):
+        c_time = ctypes.c_float(exposure_time)
+        code = self.dll.SetExposureTime(c_time)
+        self.error_check(code, 'SetExposureTime')
+
+    def set_filter_mode(self, mode):
+        c_mode = ctypes.c_int(mode)
+        code = self.dll.SetFilterMode(c_mode)
+        self.error_check(code, 'SetFilterMode')
+
+    def set_number_accumulations(self, number):
+        c_number = ctypes.c_int(number)
+        code = self.dll.SetNumberAccumulations(c_number)
+        self.error_check(code, 'SetNumberAccumulations')
+
+    def set_read_mode(self, mode):
+        code = self.dll.SetReadMode(mode)
+        self.error_check(code, 'SetReadMode')
+
+    def set_shutter(self, typ, mode, closing_time, opening_time):
+        c_typ = ctypes.c_int(typ)
+        c_mode = ctypes.c_int(mode)
+        c_closing_time = ctypes.c_int(closing_time)
+        c_opening_time = ctypes.c_int(opening_time)
+        code = self.dll.SetShutter(c_typ, c_mode, c_closing_time, c_opening_time)
+        self.error_check(code, 'SetShutter')
+
+    def set_temperature(self, temperature):
+        c_temperature = ctypes.c_int(temperature)
+        code = self.dll.SetTemperature(c_temperature)
+        self.error_check(code, 'SetTemperature')
+
+    def set_trigger_mode(self, mode):
+        c_mode = ctypes.c_int(mode)
+        code = self.dll.SetTriggerMode(c_mode)
+        self.error_check(code, 'SetTriggerMode')
+
+    def shut_down(self):
+        code = self.dll.ShutDown()
+        self.error_check(code, 'ShutDown')
+
+    def start_acquisition(self):
+        code = self.dll.StartAcquisition()
+        self.error_check(code, 'StartAcquisition')
+
+    def wait_for_acquisition(self):
+        code = self.dll.WaitForAcquisition()
+        self.error_check(code, 'WaitForAcquisition')
+
+
+class Spectrum(Parameter):
```

Comment:
I personally do not think there is a need for making this parameter a separate class. The `get_raw` method can be implemented in the driver class.

Moreover, it seems that this parameter is a good candidate for `ParameterWithSetpoints` (with `Array` validator and callable `shape`), [it is mentioned here](http://qcodes.github.io/Qcodes/examples/Creating%20Instrument%20Drivers.html?highlight=parameterwithsetpoints#Custom-Parameter-classes). I encourage you to read more about it [here](http://qcodes.github.io/Qcodes/examples/writing_drivers/Simple-Example-of-ParameterWithSetpoints.html), [here](http://qcodes.github.io/Qcodes/examples/writing_drivers/A-ParameterWithSetpoints-Example-with-Dual-Setpoints.html), [here](http://qcodes.github.io/Qcodes/examples/driver_examples/Qcodes-example-with-Signal-Hound-USB-SA124B-ParameterWithSetpoints.html), and see if you like this approach. I think that it fits perfectly here.